project/OSGi.scala

```diff
@@ -0,0 +1,87 @@
+    OsgiKeys.bundleVersion := version.value.replaceFirst("\\+", ".").replaceAll("\\+", "-")
+  )
+
+  // Need to specify scala.reflect.macros here because the bnd macro will not import this with the proper version
+  val parsing = osgiPkgHeaders(
+    exports = Seq("akka.parboiled2.*", "akka.shapeless.*", "akka.macros.*"),
+    imports = Seq("scala.reflect.macros", optionalResolution("scala.quasiquotes")))
+
+  val httpCore = osgiPkgHeaders(
+    exports = Seq("akka.http.*"),
+    imports = Seq(scalaJava8CompatImport, akkaParboiledImport, akkaShapelessImport, akkaMacrosImport))
+
+  val http = osgiPkgHeaders(
+    exports = Seq(
+      "akka.http.scaladsl.client.*",
+      "akka.http.#DSL#.server.*",
+      "akka.http.#DSL#.coding.*",
+      "akka.http.#DSL#.common.*",
+      "akka.http.#DSL#.marshalling.*",
+      "akka.http.#DSL#.unmarshalling.*"
+    ) flatMap { p =>
+      Seq(p.replace("#DSL#", "scaladsl"), p.replace("#DSL#", "javadsl"))
+    },
+    imports = Seq(scalaJava8CompatImport, akkaParboiledImport)
+  )
+
+  val httpTestkit = osgiPkgHeaders(
+    exports = Seq("akka.http.scaladsl.testkit.*", "akka.http.javadsl.testkit.*"),
+    imports = Seq("akka.testkit.*", "akka.stream.testkit.*", "org.junit") map { pkg => optionalResolution(minorVersioned(pkg)) }
+  )
+
+  val httpSprayJson = osgiPkgHeaders(exports = Seq("akka.http.scaladsl.marshallers.sprayjson"))
+
+  val httpXml = osgiPkgHeaders(exports = Seq("akka.http.scaladsl.marshallers.xml"))
+
+  val httpJackson = osgiPkgHeaders(exports = Seq("akka.http.javadsl.marshallers.jackson"))
+
+  val http2Support = osgiSettings ++ Seq(
+    OsgiKeys.importPackage := Seq(projectVersioned("akka.http.*"), akkaParboiledImport, akkaMacrosImport, akkaShapelessImport) ++ defaultImports,
+    OsgiKeys.fragmentHost := Some("com.typesafe.akka.http.core"),
+    OsgiKeys.exportPackage := Seq("""akka.http.scaladsl.*;-split-package:=merge-first"""),
+    OsgiKeys.privatePackage := Seq("akka.http.impl.*")
+  )
+
+  private def osgiPkgHeaders(exports: Seq[String] = Seq(), imports: Seq[String] = Nil) = osgiSettings ++ Seq(
+    OsgiKeys.importPackage := Seq(projectVersioned("akka.http.*")) ++ imports ++ defaultImports,
+    OsgiKeys.exportPackage := exports
+  )
+  private lazy val defaultImports = Seq("!sun.misc", akkaImport, configImport, scalaImport, "*")
+
+  private lazy val akkaParboiledImport = projectVersioned("akka.parboiled2.*")
+  private lazy val akkaShapelessImport = projectVersioned("akka.shapeless.*")
+  private lazy val akkaMacrosImport = projectVersioned("akka.macros.*")
+  private lazy val akkaImport = "akka," + patchVersioned("akka.*")
+  private lazy val configImport = patchVersioned("com.typesafe.config.*")
+  private lazy val scalaImport = minorVersioned("scala.*")
+  private lazy val scalaJava8CompatImport = """scala.compat.java8.*;version="$<range;[==,1)>""""
+
+  private def projectVersioned(pkg: String) = pkg + """;version="$<range;[===,=+);$<Bundle-Version>>""""
```

Comment:
It's the bnd tool range syntax. This allows us to specify ranges without the need to manually enter what range versions we want for a specific library. Without this, it is maddening to keep up with the version changes on every release.